src/client/Microsoft.Identity.Client/ManagedIdentity/ImdsV2ManagedIdentitySource.cs

```diff
@@ -270,7 +270,20 @@ protected override ManagedIdentityRequest CreateRequest(string resource)
 
             var clientCredentialRequestResponse = ExecuteClientCredentialRequestAsync(csrMetadata.Cuid, csr.Pem).GetAwaiter().GetResult();
 
-            throw new NotImplementedException();
+            ManagedIdentityRequest request = new(HttpMethod.Post, new Uri($"{clientCredentialRequestResponse.RegionalTokenUrl}/{clientCredentialRequestResponse.TenantId}{AcquireEntraTokenPath}"));
+
+            request.Headers.Add("x-ms-client-request-id", _requestContext.CorrelationId.ToString());
+
+            request.BodyParameters.Add("grant_type", clientCredentialRequestResponse.ClientCredential);
+            request.BodyParameters.Add("scope", "https://management.azure.com/.default");
+            if (clientCredentialRequestResponse.ClientId != null)
+            {
```

Comment:
we also need to pass the token_type = mtls_pop for the pop flow

Reply:
That's optional, and aren't we just doing bearer token flow for now?

Reply:
Yes, we do not need to write that logic, but we at least need to have the token type as one of the params in here. Or, we can maybe add a To-Do so we know this is the place we plug in the mtls logic later